nexus-watcher/src/events/moderation.rs

```diff
@@ -0,0 +1,43 @@
+use crate::events::handlers;
+use nexus_common::types::DynError;
+use pubky_app_specs::{ParsedUri, PubkyAppTag, PubkyId, Resource};
+use tracing::debug;
+
+pub struct Moderation {
+    // Moderator trusted user id
+    pub id: PubkyId,
+    // Tags to be moderated (tagged content is deleted)
+    pub tags: Vec<String>,
+}
+
+impl Moderation {
+    pub async fn should_delete(&self, tag: &PubkyAppTag, tagger_id: PubkyId) -> bool {
+        tagger_id == self.id && self.tags.contains(&tag.label)
+    }
+
+    pub async fn apply_moderation(tag: PubkyAppTag) -> Result<(), DynError> {
```

Comment:
Maybe rename `tag` to `moderaton_tag` or `mod_tag`, to avoid confusion with the moderated tag in the `Resource::Tag` branch below.